### PR TITLE
Comma separated arrays

### PR DESCRIPTION
**Pull request by mihai-chiorean**
array parameters now work as comma separated. the framework transforms the string to an array. array should not be empty or validate fails. also, removes any trailing commas. Reviewers: @seandunford

### CODE REVIEW

File: lib/createParamObjects.js [hunk 1]
```diff
@@ -21,8 +21,19 @@ function createParamObjects(params, ParamType) {
         } else if (type === Boolean) {
             // TODO: we should convert the output type to a boolean
         } else if (_.isArray(type) || type === Array) {
-            return function (val) {
-                assert.ok(_.isArray(val), sprintf(validationErrorTemplate, name, "Array"));
+            return function (aVal) {
+                var value = aVal;
+                if(typeof aVal === 'string') {
+                    var v = aVal.trim();
+                    if(v[0] == ',') {
```
Comment (SeanDunford) [inline lib/createParamObjects.js]: Are you sure that you need this processing? Split should just take care of it. Worst case you get an empty val in the array that is just going to get ignored anyway.

File: lib/createParamObjects.js [hunk 2]
```diff
@@ -51,17 +61,38 @@ function createParamObjects(params, ParamType) {
...
         } else if(type == String) {
-	    return function(val) {
-		if(val == 'undefined' || val == 'null') {
-		    return null;
-		}
-		return val;
-	    }
-	}
-        return;
+            return function(val) {
```
Comment (SeanDunford) [inline lib/createParamObjects.js]: Duplicated code from above.

**Comment by SeanDunford** [comment]
+1

File: lib/createParamObjects.js (hunk 2 above)
Comment (mihai-chiorean) [inline lib/createParamObjects.js]: true that. abstracting it

File: lib/createParamObjects.js (hunk 1 above)
Comment (mihai-chiorean) [inline lib/createParamObjects.js]: I'd rather have it for safety. it shouldn't be too expensive anyway

Comment (SeanDunford) [inline lib/createParamObjects.js]: √

**Pull request merged**